code/game/objects/structures/window.dm

```diff
@@ -396,9 +396,10 @@
 //If you find this like 4 years later and construction still hasn't been refactored, I'm so sorry for this //Adding a timestamp, I found this in 2020, I hope it's from this year -Lemon
 //2021 AND STILLLL GOING STRONG
 /obj/structure/window/reinforced/attackby(obj/item/I, mob/living/user, params)
+	var/list/modifiers = params2list(params)
 	switch(state)
 		if(RWINDOW_SECURE)
-			if(I.tool_behaviour == TOOL_WELDER && user.combat_mode)
+			if(I.tool_behaviour == TOOL_WELDER && modifiers && modifiers["right"])
```

Comment:
Please move this to `attackby_secondary`.

Reply:
And if it's not a welder, cancel the attack chain (and perhaps even give them feedback saying to use a welder if it has some other tool behavior).

Reply:
Do you think that all deconstruction steps should be moved to right click and leave left click as attack? When testing this I realized how weird it is that only the first step has to be in a different mode.